pkcs11/src/new/functions/slot_token_management.rs

```diff
@@ -1,13 +1,63 @@
-use crate::new::types::function::Result;
+use crate::new::types::function::Rv;
 use crate::new::types::slot_token::Slot;
 use crate::new::Pkcs11;
+use crate::new::Result;
+use std::convert::TryInto;
 
 impl Pkcs11 {
     pub fn get_slots_with_token(&self) -> Result<Vec<Slot>> {
-        Ok(Vec::new())
+        let mut slot_count = 0;
+
+        Rv::from(unsafe {
+            ((*self.function_list).C_GetSlotList.unwrap())(
+                pkcs11_sys::CK_TRUE,
+                std::ptr::null_mut(),
```

Comment:
Same here? Does this need abstracting in the future?

Reply:
This one is OK! The first call this parameter is set to `NULL` just to get the number of slot, written by the API in `slot_count`